blockchain/blockdao/blockdao.go

```diff
@@ -105,37 +105,40 @@ func (dao *blockDAO) Start(ctx context.Context) error {
 	return dao.checkIndexers(ctx)
 }
 
-func (dao *blockDAO) fillWithBlockInfoAsTip(ctx context.Context, height uint64) (context.Context, error) {
-	g, ok := genesis.ExtractGenesisContext(ctx)
-	if !ok {
-		return nil, errors.New("failed to find blockchain ctx")
+func (dao *blockDAO) fillWithBlockInfoAsTip(ctx context.Context, chainID uint32, height uint64) (context.Context, error) {
+	bcCtx := protocol.BlockchainCtx{
+		Tip: protocol.TipInfo{
+			Height: height,
+		},
+		ChainID: chainID,
 	}
-	bcCtx := protocol.BlockchainCtx{}
```

Comment:
`bcCtx := protocol.GetBlockchainCtx(ctx)`
is this the only line we need to change?

Reply:
the current change is more efficient

Reply:
can actually remove fillWithBlockInfoAsTip, see 2nd commit